TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/QueryableExtensions.cs

```diff
@@ -0,0 +1,71 @@
+using System.Collections;
+using System.Linq.Expressions;
+
+namespace TeachingRecordSystem.SupportUi;
+
+public static class QueryableExtensions
+{
+    public static IOrderedQueryable<TSource> OrderBy<TSource, TKey>(
+        this IQueryable<TSource> source,
+        SortDirection direction,
+        Expression<Func<TSource, TKey>> keySelector)
+    {
+        return direction == SortDirection.Ascending ? source.OrderBy(keySelector) : source.OrderByDescending(keySelector);
+    }
+
+    public static IOrderedQueryable<TSource> ThenBy<TSource, TKey>(
+        this IOrderedQueryable<TSource> source,
+        SortDirection direction,
+        Expression<Func<TSource, TKey>> keySelector)
+    {
+        return direction == SortDirection.Ascending ? source.ThenBy(keySelector) : source.ThenByDescending(keySelector);
+    }
+
+    public static async Task<ResultPage<T>> GetPageAsync<T>(
+        this IQueryable<T> source,
+        int? currentPage,
+        int itemsPerPage)
+    {
+        var page = Math.Max(currentPage ?? 1, 1);
+
+        var total = await source.CountAsync();
+
+        var lastPage = ResultPage.GetLastPage(total, itemsPerPage);
+
+        // Is the page requested out of bounds?
+        if (page > lastPage)
+        {
+            page = lastPage;
+        }
```

Comment:
This logic felt cleaner to me when it was all part of a single `Pagination` class, now it's split across many classes

Reply:
>now it's split across many classes

How so?

Reply:
The `ResultsPage<T>` classes and the extension method class